sakura/sakura/pubsub/rabbitmq/pubsub_client.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+from collections.abc import Coroutine
+
+from sakura.exceptions import SubscriberNotFoundError
+from sakura.pubsub.client import PubSubClient
+from sakura.pubsub.rabbitmq.rabbitmq_client import RabbitMQClient
+from sakura.pubsub.rabbitmq.rabbitmq_client import Settings as RabbitMQClientSettings
+from sakura.pubsub.rabbitmq.settings import Settings
+from sakura.pubsub.rabbitmq.subsriber import RabbitMQSubscriber
+
+
+class RabbitMQPubSubClient(PubSubClient):
+    Settings = Settings
+    __running_subscribers: dict[str, RabbitMQSubscriber] = {}
+
+    def __init__(self, settings: Settings):
+        self.__settings = settings
+        self.serialization = settings.serialization
+        self.subscribers = settings.subscribers
+        self.publishers = settings.publishers
+
+        self.__rabbitmq_client = RabbitMQClient(RabbitMQClientSettings.from_dynaconf(raw_settings=settings.params))
+        super().__init__()
+
+    def setup(self):
+        loop = asyncio.get_event_loop()
+        loop.run_until_complete(self.__rabbitmq_client.setup())
```

Comment:
Can we not use run_until_complete here?

Reply:
Maybe using your "new" async classmethod, but the setup has to be async since it's initializing the different queues and exchanges over rabbitmq